openmetadata-ui/src/main/resources/ui/src/pages/LogsViewer/LogsViewer.component.tsx

```diff
@@ -0,0 +1,312 @@
+/*
+ *  Copyright 2022 Collate
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+import { Button, Card, Col, Row, Space, Typography } from 'antd';
+import { AxiosError, AxiosResponse } from 'axios';
+import { isEmpty, isNil, isUndefined, toNumber } from 'lodash';
+import React, {
+  Fragment,
+  useEffect,
+  useLayoutEffect,
+  useMemo,
+  useState,
+} from 'react';
+import { useTranslation } from 'react-i18next';
+import { LazyLog } from 'react-lazylog';
+import { useParams } from 'react-router-dom';
+import {
+  getIngestionPipelineByName,
+  getIngestionPipelineLogById,
+} from '../../axiosAPIs/ingestionPipelineAPI';
+import { CopyToClipboardButton } from '../../components/buttons/CopyToClipboardButton/CopyToClipboardButton';
+import TitleBreadcrumb from '../../components/common/title-breadcrumb/title-breadcrumb.component';
+import Loader from '../../components/Loader/Loader';
+import { PipelineType } from '../../generated/api/services/ingestionPipelines/createIngestionPipeline';
+import { IngestionPipeline } from '../../generated/entity/services/ingestionPipelines/ingestionPipeline';
+import { Paging } from '../../generated/type/paging';
+import { getIngestionStatuses } from '../../utils/CommonUtils';
+import { getLogBreadCrumbs } from '../../utils/LogsViewer.utils';
+import { showErrorToast } from '../../utils/ToastUtils';
+import LogViewerSkeleton from './LogsViewer-skeleton.component';
+import {
+  IngestionPipelineLogByIdInterface,
+  LogViewerParams,
+} from './LogsViewer.interfaces';
+import './LogsViewer.style.less';
+
+const LogsViewer = () => {
+  const { logEntityType, ingestionName } = useParams<LogViewerParams>();
+
+  const { t } = useTranslation();
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [logs, setLogs] = useState<string>('');
+  const [ingestionDetails, setIngestionDetails] = useState<IngestionPipeline>();
+  const [paging, setPaging] = useState<Paging>();
+
+  const fetchLogs = async (
+    ingestionId?: string,
+    pipelineType?: PipelineType
+  ) => {
+    try {
+      const res = (await getIngestionPipelineLogById(
+        ingestionId || ingestionDetails?.id || '',
+        paging?.total !== paging?.after ? paging?.after : ''
+      )) as AxiosResponse<IngestionPipelineLogByIdInterface>;
```

Comment:
Type should be identify from the API function and shouldn't mention as overlap here

Reply:
@chirag-madlani , Did not get this? can you suggest changes?